inspirehep/modules/records/receivers.py

```diff
@@ -142,10 +145,11 @@ def push_to_orcid(sender, record, *args, **kwargs):
 
     task_name = current_app.config['ORCID_PUSH_TASK_ENDPOINT']
 
-    for orcid in get_orcids_for_push(record):
-        token = get_push_access_token(orcid)
-        if token is None:
-            continue
+    orcids = get_orcids_for_push(record)
```

Comment:
This code produces the same results as the old one, it just leverages the function `get_push_access_tokens` which is an optimized query.
Thus the old tests (the tests starting with `test_orcid_push*` here: https://github.com/inspirehep/inspire-next/blob/master/tests/integration/records/test_records_receivers.py#L262) exercise this new code.